### PR TITLE
Chores/audit log UI

### LINKED ISSUE #1887: Paging audit log just increases length of the page rather than changing to new page
Note that I can't reproduce this in production but I can reproduce it using the main branch on my local machine. 

gz#376

### LINKED ISSUE #1920: Audit Log filters cant be reset
**Describe the bug**
Once you select an audit log environment filter, its impossible to deselect it without leaving the page and returning

**To Reproduce**
Go to audit log, click filter, click filter again

**Expected behavior**
The second click deselects the filter

**Screenshots**

**How are you running Flagsmith?**
- [ ] Self Hosted with Docker
- [ ] Self Hosted with Kubernetes
- [X] SaaS at flagsmith.com
- [ ] Some other way

**Additional context**

gz#417

### LINKED ISSUE #1921: Audit logs should show the environment they are owned by
**Is your feature request related to a problem? Please describe.**
The Audit log list doenst show the environment a log is associated with

**Describe the solution you'd like**
Show the environment in the list

**Describe alternatives you've considered**
None

**Additional context**
None

gz#418

### LINKED ISSUE #1964: Audit log environment filter doesnt alter UI filter buttons
**Describe the bug**
Audit log environment filter doesnt alter UI filter buttons

**To Reproduce**

1. Go to audit log
2. Filter on environment 
3. Button doesnt change appearance

**Expected behavior**
The filter is applied and the filter button changes colour to show it is enabled

**How are you running Flagsmith?**
- [ ] Self Hosted with Docker
- [ ] Self Hosted with Kubernetes
- [x] SaaS at flagsmith.com
- [ ] Some other way

gz#455

### PR DESCRIPTION
**Pull request by kyle-ssg**
Thanks for submitting a PR! Please check the boxes below:

- [x] I have run [`pre-commit`](https://docs.flagsmith.com/deployment/locally-api#pre-commit) to check linting
- [x] I have filled in the "Changes" section below?
- [x] I have filled in the "How did you test this code" section below?

## Changes

Refactors the audit log, also fixes https://github.com/Flagsmith/flagsmith/issues/1921 https://github.com/Flagsmith/flagsmith/issues/1920 https://github.com/Flagsmith/flagsmith/issues/1827

image
image

### CODE REVIEW

File: frontend/common/types/responses.ts [hunk 1]
```diff
@@ -61,9 +61,10 @@ export type AuditLogItem = {
   environment?: Environment;
   project: ProjectSummary;
   related_object_id: number;
-  related_object_type: string;
+  related_object_type: "FEATURE" | "FEATURE_STATE" | "ENVIRONMENT" | "CHANGE_REQUEST";
```
Comment (matthewelwell) [inline frontend/common/types/responses.ts]: This isn't actually the full list, it's missing `"SEGMENT"`, although we may add new ones in the future. 

https://github.com/Flagsmith/flagsmith/blob/main/api/audit/related_object_type.py

Comment (kyle-ssg) [reply frontend/common/types/responses.ts]: Cool, will add. I was considering using this info for display but I don't think I will.

**Comment by dabeeeenster** [comment]
This looks good. I wonder whether you should be able to filter on > 1 environment tho? At the moment selecting a different env deselects the currently selected one. That removes the ability to filter on > 1

Also the mouse over shows a hover link underline on the env in the table which I think needs removing - we don't have underlines anywhere AFAIK

image

File: frontend/common/types/responses.ts (hunk 1 above)
Comment (matthewelwell) [reply frontend/common/types/responses.ts]: Not sure if you haven't pushed yet or I just didn't understand the above but `"SEGMENT"` is still missing here?

**Comment by matthewelwell** [comment]
>I wonder whether you should be able to filter on > 1 environment tho? At the moment selecting a different env deselects the currently selected one. That removes the ability to filter on > 1

This is outstanding from @dabeeeenster's feedback but I would argue it's not really necessary. Thoughts @dabeeeenster?

**Comment by dabeeeenster** [comment]
Happy to merge this without being able to select multiple envs

**Review by matthewelwell — APPROVED** [review]

**Pull request merged**